### PR TITLE
MM-11398 Expand extension section, add embed notes

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: site/content/extend/getting-started/_index.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Mattermost is a great communication platform but sometimes it's more convenient to send and receive messages from within a non-Mattermost application. With Mattermost's rich RESTful web service API and OAuth 2.0 support it's possible to build your own clients to embed Mattermost within any application of your choice.
+
+[Read more about embedding.](/extend/embed)
+
+### Fork
+
+Mattermost is open-source, which means you have the ability to fork the source code if you need to modify Mattermost in ways not otherwise supported.
+
+[Read about licensing around forking Mattermost.](https://github.com/mattermost/mattermost-server/blob/master/LICENSE.txt)
```
Comment (jasonblais) [inline site/content/extend/getting-started/_index.md]: completely 0/5, but we also have this FAQ that might be easier to read https://docs.mattermost.com/overview/faq.html#how-can-i-create-a-derivative-work-of-mattermost-as-my-own-commercial-solution

File: site/content/extend/getting-started/_index.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Getting Started
+
+Mattermost can be extended by building and installing plugins, by embedding Mattermost in another application, or by forking the codebase.
+
+## Extension Methods
+
+### Plugins (Beta)
+
+Plugins are a powerful way to add new functionality and UI to Mattermost - in an easy to maintain and update way. They are easy to build and give you more power than any other integration method.
```
Comment (jasonblais) [inline site/content/extend/getting-started/_index.md]: How about: `Plugins are easy to build, effortless to maintain, and offer powerful ways to add new functionality and UI to both the server and web/desktop apps, not available with any other integration method.`

File: site/content/extend/embed/_index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+With Mattermost's rich RESTful web service API and OAuth 2.0 support it's possible to build your own clients to embed Mattermost within any application of your choice.
+
+Questions about embedding Mattermost? Ask in the [Developer Toolkit channel](https://pre-release.mattermost.com/core/channels/developer-toolkit) on our Mattermost community instance.
+
+## Embedding Methods
+
+### Browser Extensions and Add-Ons
+
+Many modern browsers have support for extensions or add-ons that would allow you to build and embed a Mattermost client on web pages of your choice. For Chrome there are [extensions](https://developer.chrome.com/extensions), Firefox has [add-ons](https://addons.mozilla.org/en-US/developers/), and IE/Edge have [extensions](https://docs.microsoft.com/en-us/microsoft-edge/extensions).
```
Comment (jasonblais) [inline site/content/extend/embed/_index.md]: `Chrome has [extensions](...) ...` for consistent sentence structure.

**Comment by jwilander** [comment]
@jasonblais updated

**Pull request merged**